### DISCUSSION TITLE
Multi arity functions

### DISCUSSION THREAD

**Discussion by madhat2r** [hylang/hy#2570] [Q&A] [answered]
Does hy support multi arity functions? If so can someone point me in the right direction please? Thanks.

**Comment by Kodiologist** [comment, marked as answer]
They were removed in #1842. Use [multidispatch from PyPI](https://pypi.org/project/multipledispatch/) instead.

Unless by "multi-arity" you just mean "variadic". That's accomplished with optional arguments, `#*`, and `#**`; see the documentation of `defn`.
